src/notification/NotificationController.ts

```diff
@@ -0,0 +1,86 @@
+import BaseController, { BaseConfig, BaseState } from '../BaseController';
+
+interface viewedNotification {
+[id: number]: boolean;
+}
+
+export interface Notification{
+    id: number;
+    title: string;
+    description: string;
+    date: string;
+    image?: string;
+    actionText: string;
+    isShown?: boolean;
+}
+
+// NotitificationConfig will hold the notifications from JSON file read from `metamask-extension`
+export interface NotificationConfig extends BaseConfig{
+    notificationsFromFile: { [whatsnew: string]: Notification[]};
+}
+
+// Notification state will hold all the seen and unseen notifications
+// that are still active
+export interface NotificationState extends BaseState{
+    notifications: Notification[];
+}
+
+const defaultState = { notifications: [] };
+
+/**
+ * Controller for managing the notifications for new features introducesd into the applications
+ */
+export class NotificationController extends BaseController<NotificationConfig, NotificationState> {
+
+    constructor(config: NotificationConfig, state?: NotificationState) {
+        const { notificationsFromFile } = config;
+        super(config, state || defaultState);
+        this.initialize();
+        this.addNotifications(notificationsFromFile.whatsnew);
+    }
+
+    /**
+     * Compares the notifications in teh states with the notifications from the files to check if there is any new notitifcations/announcements
+     * if yes, the new notification will be added to the state with a flag indicating that the notification is not seen by the user.
+     * @param filedNotifications
```

Comment:
I think this was meant to say `notificationsFromFile` or something along those lines? :thinking:

On the subject of the `notificationsFromFile` name, maybe it would be simpler to call it `notifications`? Or `defaultNotifications`? It's not especially important that they come from a file, and it might not even be true all of the time (it had been suggested that we could source them from a remote service).